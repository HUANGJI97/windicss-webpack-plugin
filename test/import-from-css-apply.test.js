"use strict";

const getModuleSource = require('./helpers/getModuleSource')
const vueWebpackCompiler = require ('./helpers/vueWebpackCompiler')

describe("@import css with @apply", function() {

  it('should render vue', done => {
    const compiler = vueWebpackCompiler('import-css-with-apply', )
    compiler.run((err, stats) => {
      expect(stats.compilation.errors).toStrictEqual([])

      const toNotTransformCss = getModuleSource('global.css', stats)
      expect(toNotTransformCss).not.toContain('@apply')
      expect(toNotTransformCss).toMatchSnapshot('@apply')

      done(err)
    });
  });
});
