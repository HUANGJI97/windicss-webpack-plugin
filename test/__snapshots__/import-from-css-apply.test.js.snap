// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@import css with @apply should render vue: @apply 1`] = `
"// Imports
import ___CSS_LOADER_API_IMPORT___ from \\"../../../../node_modules/css-loader/dist/runtime/api.js\\";
var ___CSS_LOADER_EXPORT___ = ___CSS_LOADER_API_IMPORT___(function(i){return i[1]});
// Module
___CSS_LOADER_EXPORT___.push([module.id, \\"h1 {\\\\n  font-size: 2.25rem;\\\\n  line-height: 2.5rem;\\\\n}\\\\nh2 {\\\\n  font-size: 1.875rem;\\\\n  line-height: 2.25rem;\\\\n}\\\\nh3 {\\\\n  font-size: 1.5rem;\\\\n  line-height: 2rem;\\\\n}\\", \\"\\"]);
// Exports
export default ___CSS_LOADER_EXPORT___;
"
`;
